src/Tokenizer/Tokens.php

```diff
@@ -321,18 +334,27 @@ public function offsetUnset($index): void
         }
 
         parent::offsetUnset($index);
+
+        if (\count($this) - 1 === $index) {
+            $this->setSize(\count($this) - 1);
+        }
     }
 
     /**
      * Set collection item.
      *
-     * Warning! `$newval` must not be typehinted to be compatible with `ArrayAccess::offsetSet` method.
```

Comment:
why dropping this ?

Reply:
typehint is already there

Reply:
unfortunately it is not
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7698/files#diff-f0f05b598c6bff06cf727fc98b5d1191f18793eb4f9dce0c61b7d93ec8a7a334R349

I added this comment because getting poked by SCA again and again to add typehint for that param in method prototype

Reply:
What SCA you mean? The `@param` is present for both params, PHPStan can detect all issues and for `$index` we have the same situation /wo comment.

Reply:
there is no type in `public function offsetSet($index, $newval): void` so I want to keep that comment. 
not sure if it's worth to argue for this for 2 weeks

Reply:
we argue in this PR since January :(

Reply:
maybe just easier to not remove this line.